[JENKINS-53055] Upgrade libpam4j dependency

jvz opened the pull request:
See [JENKINS-53055](https://issues.jenkins-ci.org/browse/JENKINS-53055).

The underlying code in libpam4j has remained very stable and backwards compatible and requires no additional tests written. See [libpam4j changes since 1.8](https://github.com/kohsuke/libpam4j/compare/libpam4j-1.8...libpam4j-1.11).

### Proposed changelog entries

* `Internal:` upgrade libpam4j to 1.11

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@reviewbybees @daniel-beck @Wadeck @oleg-nenashev @jeffret-b

jvz committed:
Upgrade to libpam4j 1.11

oleg-nenashev reviewed (dismissed):
It seems that there is no changelog for this lib :(

jvz commented:
@daniel-beck good catch! I'll update this.

jvz committed:
Remove libpam4j from core dependencies

This dependency is no longer used ever since PAM authn/authz support
was extracted into the pam-auth-plugin years ago. This reduces the need
to keep security libraries up to date in multiple places.

daniel-beck commented:
FWIW using a locally patched https://github.com/jenkins-infra/deprecated-usage-in-plugins I scanned all plugins' latest releases for use of `org.jvnet.libpam` and the only hit was `pam-auth` plugin (which bundles it anyway).

jvz commented:
@oleg-nenashev unfortunately, no, there is no changelog for libpam4j, but the commit history is very short, and the diffs between versions have mainly been JNA upgrades or other dependencies.

jvz committed:
Revert "Remove libpam4j from core dependencies"

This reverts commit a7c7a52a4875b1f6248a4bffb0abd05431b1cbcb.

jeffret-b reviewed (approved):
Given the little tentacles of this library still left around, I agree that it's probably better to just upgrade the version and keep it at this point.

jvz commented:
Oh hey, turns out that JNA and JNR don't work too well with Java 9 for some reason. If that never gets improved, we might just want to use JNI instead as that will never go away (it's how a lot of the standard Java class library is implemented; see any method marked `native`).
